ros2でclang-tidyを実行するときstd library周りでエラーが出る

<!-- 不具合のテンプレート -->
**概要**

clang-tidy 15以前だと以下のような出力がされる
```bash
error: too many errors emitted, stopping now [clang-diagnostic-error]
/opt/ros/humble/include/rclcpp/rclcpp/client.hpp:550:10: error: no member named 'optional' in namespace 'std' [clang-diagnostic-error]
    std::optional<CallbackInfoVariant>
    ~~~~~^
/opt/ros/humble/include/rclcpp/rclcpp/client.hpp:550:19: error: use of undeclared identifier 'CallbackInfoVariant' [clang-diagnostic-error]
    std::optional<CallbackInfoVariant>
                  ^
/opt/ros/humble/include/rclcpp/rclcpp/client.hpp:551:5: error: use of undeclared identifier 'optional_pending_request' [clang-diagnostic-error]
    optional_pending_request = this->get_and_erase_pending_request(request_header->sequence_number);
    ^
/opt/ros/humble/include/rclcpp/rclcpp/client.hpp:552:10: error: use of undeclared identifier 'optional_pending_request' [clang-diagnostic-error]
    if (!optional_pending_request) {
         ^
/opt/ros/humble/include/rclcpp/rclcpp/client.hpp:555:21: error: use of undeclared identifier 'optional_pending_request' [clang-diagnostic-error]
    auto & value = *optional_pending_request;
                    ^
/opt/ros/humble/include/rclcpp/rclcpp/client.hpp:558:14: error: no member named 'holds_alternative' in namespace 'std' [clang-diagnostic-error]
    if (std::holds_alternative<Promise>(value)) {
        ~~~~~^
/opt/ros/humble/include/rclcpp/rclcpp/client.hpp:558:32: error: unexpected type name 'Promise': expected expression [clang-diagnostic-error]
    if (std::holds_alternative<Promise>(value)) {
                               ^
/opt/ros/humble/include/rclcpp/rclcpp/client.hpp:561:21: error: no member named 'holds_alternative' in namespace 'std' [clang-diagnostic-error]
...
```

`/opt/ros/*`や`/usr/*`にあるものでそのリポジトリのコードを修正すべきということではないので無視するか、
出ないようにする必要があるが、
これらは構文解析用にASTを作る際に出力されるエラーのようでclang-tidyで無視するオプションはない。

clang-tidy 16に上げることで出なくなるのでclang-tidy 16を使うようにしたい

動作検証したCIは[こちら](https://github.com/sbgisen/cube/actions/runs/5595023825/jobs/10230597439)

**再現手順**
**修正しないとどう困るか**
**原因**
**修正案**

--- a/.github/workflows/ros2-test.yml
+++ b/.github/workflows/ros2-test.yml
@@ -171,15 +171,12 @@ jobs:
         run: |
           wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key|sudo apt-key add -
           sudo add-apt-repository -y \
-          "deb http://apt.llvm.org/$(lsb_release -cs)/ llvm-toolchain-$(lsb_release -cs)-13 main"
+          "deb http://apt.llvm.org/$(lsb_release -cs)/ llvm-toolchain-$(lsb_release -cs)-16 main"
           sudo apt-get update
-          sudo apt-get install -y clang-tidy-13
-          pip3 install compdb
-          cat ../../build/*/compile_commands.json > ../../build/compile_commands.json
-          sed -i -e ':a;N;$!ba;s/\]\n*\[/,/g' ../../build/compile_commands.json
-          compdb -p ../../build list > ../../compile_commands.json
+          sudo apt-get install -y clang-tidy-16
+          cp ../../build/compile_commands.json ../../compile_commands.json
           echo "------------------------ Run Clang-tidy ------------------------"
-          result=`clang-tidy-13 --quiet ${{ steps.changes.outputs.cpp_files }} \
+          result=`clang-tidy-16 --quiet ${{ steps.changes.outputs.cpp_files }} \
           --header-filter="${{ github.workspace }}/ros/src/${{ github.event.repository.name }}/include/.*"`
           if [[ -z ${result} ]]; then
             exit 0

